### DISCUSSION TITLE
json/jsonb functions and operators

### DISCUSSION THREAD

**Discussion by gkaracha** [circuithub/rel8#314] [Q&A] [answered]
Hello! I am migrating some queries to Rel8, and I find myself in need of [operators that deal with jsonb data](https://www.postgresql.org/docs/9.5/functions-json.html). In particular `#>>` (or `->` and `->>` would also do) and `jsonb_array_elements`. I haven't found anything about them in the Rel8 documentation. Are they supported at the moment in any form? Cheers!

**Comment by sfwc** [comment]
I don't think there's built-in support for this as of 1.4.1, maybe it's been added more recently? But there's always `binaryOperator`, e.g.

    objectLookup :: Expr (JSONBEncoded Aeson.Value) -> Expr Text -> Expr (Maybe Text)
    objectLookup = binaryOperator "->>"

**Reply by gkaracha** [reply]
Thanks! Yeah, I did manage to get the basic binary operators using `binaryOperator` indeed, but `jsonb_array_elements` is still giving me grief. It's impossible to ascribe it a proper type, currently, but it seems there has been progress on that front: https://github.com/circuithub/rel8/blob/77fb0a3262a3c5ca9f7b55223c715364e16f4cdc/src/Rel8/Query/Function.hs#L25 Hopefully it will be possible to do so with the next release

**Comment by gkaracha** [comment, marked as answer]
Indeed with the latest release one can simply do the following so I think we can close this :)
```haskell
arrayElements :: Expr (Maybe Value) -> Query (Expr Value)
arrayElements = queryFunction "jsonb_array_elements"
```
